### PR TITLE
untildify path for creating template

### LINKED ISSUE #7366: sls create --path param does not work with tilde for home directory
# Bug Report

## Description

1. What did you do? Ran `serverless create --template aws-nodejs --path ~/code/serverless-rest-api`
1. What happened? The project was created at `${CURRENT_DIRECTORY}/Users/${USER}/code/serverless-rest-api`
1. What should've happened? The project should have been created at `/Users/${USER}/code/serverless-rest-api`
1. What's the content of your `serverless.yml` file? Not applicable
1. What's the output you get when you use the `SLS_DEBUG=*` environment variable (e.g. `SLS_DEBUG=* serverless deploy`)

Here's the output for the create command:

```
jrthomerson@mine /tmp
$ serverless create --template aws-nodejs --path ~/code/serverless-rest-api
Serverless: Generating boilerplate...
Serverless: Generating boilerplate in "/private/tmp/Users/jrthomerson/code/serverless-rest-api"
 _______                             __
|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|__|  \___/|_____|__| |__|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v1.63.0
 -------'

Serverless: Successfully generated boilerplate for template: "aws-nodejs"
```

The debug output won't be much more helpful, but since you're asking for it:

```
jrthomerson@mine /tmp
$ SLS_DEBUG=* serverless create --template aws-nodejs --path ~/code/serverless-rest-api
Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command config:tabcompletion
Serverless: Load command config:tabcompletion:install
Serverless: Load command config:tabcompletion:uninstall
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Load command output
Serverless: Load command output:get
Serverless: Load command output:list
Serverless: Load command param
Serverless: Load command param:get
Serverless: Load command param:list
Serverless: Invoke create
Serverless: Generating boilerplate...
Serverless: Generating boilerplate in "/private/tmp/Users/jrthomerson/code/serverless-rest-api"
 _______                             __
|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|__|  \___/|_____|__| |__|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v1.63.0
 -------'

Serverless: Successfully generated boilerplate for template: "aws-nodejs"
```

Note: this is OSX. Versions:

```
jrthomerson@mine /tmp
$ node --version
v12.13.1
jrthomerson@mine /tmp
$ sls --version
Framework Core: 1.63.0
Plugin: 3.3.0
SDK: 2.3.0
Components Core: 1.1.2
Components CLI: 1.4.0
```

Similar or dependent issues:
None that I know of.

### PR DESCRIPTION
**Pull request by yyylksdy**
## What did you implement
untildify path for absolute path resolution

Closes #7366 

## How can we verify it
run serverless create --template aws-nodejs --path ~/code/serverless-rest-api

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: lib/plugins/create/create.js [hunk 1]
```diff
@@ -223,7 +223,7 @@ class Create {
 
     // create (if not yet present) and chdir into the directory for the service
     if (boilerplatePath) {
-      const newPath = path.join(process.cwd(), boilerplatePath);
+      const newPath = untildify(boilerplatePath);
```
[the remark this answers is not in the document]
Comment (yyylksdy) [reply lib/plugins/create/create.js]: cool, I've used path. resolve , i tested both relative path and  absolute path are ok to run.

File: lib/plugins/create/create.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ class Create {
     } else if ('template-path' in this.options) {
       // Copying template from a local directory
       const servicePath = this.options.path
-        ? untildify(this.options.path)
+        ? path.resolve(this.options.path)
```
Comment (medikoo) [inline lib/plugins/create/create.js]:
```suggestion
        ? path.resolve(process.cwd(), untildify(this.options.path))
```

File: lib/plugins/create/create.js [hunk 3]
```diff
@@ -223,7 +223,7 @@ class Create {
 
     // create (if not yet present) and chdir into the directory for the service
     if (boilerplatePath) {
-      const newPath = path.join(process.cwd(), boilerplatePath);
+      const newPath = path.resolve(boilerplatePath);
```
Comment (medikoo) [inline lib/plugins/create/create.js]:
```suggestion
      const newPath = path.resolve(process.cwd(), untildify(boilerplatePath));
```

File: lib/plugins/create/create.js (hunk 2 above)
Comment (yyylksdy) [reply lib/plugins/create/create.js]: thank u for you suggestions. i tested ~/, it works well when i use path.resolve

File: lib/plugins/create/create.js (hunk 3 above)
Comment (yyylksdy) [reply lib/plugins/create/create.js]: ok. i will change that. but i definitely tested ~/

Comment (medikoo) [reply lib/plugins/create/create.js]: `path.resolve` doesn't expand `~/` you may simply confirm it in Node.js console.

If it worked for you it means some other code path was followed, where `untildify` was involved

Comment (yyylksdy) [reply lib/plugins/create/create.js]: got u. could u pls check if we can merge this pr now ?

**Comment by yyylksdy** [comment]
> @yyylksdy looks great! It'll be nice thought to introduce some regression test.
> 
> I think it should be straightforward. Just add unit test that involves path starting with `~/` at `path` option, and confirm that `serverless.config.servicePath` is as expected (note that in tests home dir is mocked to be in some temp dir, so being it at temp dir would be expected)

Hey,  is the unit test ok to go ?

**Comment by medikoo** [comment]
> Hey, is the unit test ok to go ?

Sorry for delay. I didn't look as review was not re-requested. Looking into it now!

**Pull request merged**